core/src/main/java/com/datastax/oss/driver/internal/core/metadata/schema/parsing/DataTypeCqlNameParser.java

```diff
@@ -69,8 +69,23 @@ private DataType parse(
       return nativeType;
     }
 
+    if (parser.isEOS()) {
+      // No parameters => it's a UDT
+      CqlIdentifier name = CqlIdentifier.fromCql(type);
```

Comment:
It's simpler to move UDT before collections.
Note that this would accept a UDT called `set`, but it's not a big deal since that's impossible to create on the server anyway.